### DISCUSSION TITLE
Restarting kernel...

### DISCUSSION THREAD

**Discussion by auesro** [brianhie/scanorama#121] [Q&A] [answered]
I am trying to `correct_scanpy` a dataset made of 4 batches with 32269 cells and 23546 genes in total. I am running into a "Restarting kernel..." error in Spyder that stops everything:

```
Python 3.7.12 | packaged by conda-forge | (default, Oct 26 2021, 06:08:53) 
Type "copyright", "credits" or "license" for more information.

IPython 7.32.0 -- An enhanced Interactive Python.

runcell(0, '/home/auesro/Desktop/Cell_Ranger/Merge/AER_Merge_v2.py')

runcell('SCANORAMA CORRECTION IN ALL GENES', '/home/auesro/Desktop/Cell_Ranger/Merge/AER_Merge_v2.py')
Found 23546 genes among all datasets
[[0.         0.830794   0.82057985 0.65576613]
 [0.         0.         0.8549192  0.69690722]
 [0.         0.         0.         0.79336977]
 [0.         0.         0.         0.        ]]
Processing datasets (1, 2)
Processing datasets (0, 1)
Processing datasets (0, 2)


Restarting kernel...
```
The workstation has 64GB RAM and 16 CPUs.

Any ideas?

**Comment by brianhie** [comment, marked as answer]
Hi @auesro, I'm not sure what's happening, and without an error message it's a bit difficult to tell what's happening. Can you try integrate_scanpy() instead? That method definitely integrates better.

**Comment by auesro** [comment]
Hi @brianhie,

I tottally agree, the lack of error message makes it hard to solve. It must be something to do with the available RAM because when I subset the dataset to less cells, it runs fine but...its not a crazy ammount of cells to begin with.

Regarding the use of corrected vs integrate, is there any scenario in which you would recommend having the data matrix corrected?

Thanks!

A

**Reply by brianhie** [reply]
I would definitely recommend just integrate(), as it appears to benchmark better and since I would not recommend doing analysis on the modified full data matrix anyways.

**Reply by auesro** [reply]
Sounds good!

Just to add to my original issue: it is a matter of RAM, the dataset is fully analyzed with correct_scanpy() in a different workstation with a lot more RAM available.

Thanks!
